### PR TITLE
Feature/18 refactor context

### PR DESCRIPTION
**Pull request by Ameer-Alaswad**
Refactor app context

- Changed context and some vaiables names to better ones.
- Fixes a bug in navbar that was causing the navbar not to render when the user is not signedIn.
- Adds parseLocalStorage helper function && adds AppContextData.tsx.
- Organized the code for better readability and maintainability.

### CODE REVIEW

File: frontend/src/hooks/useAppContext.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        isMenuOpen,
+        isMobileMenuOpen,
+        handleMobileMenuClose,
+        handleMenuClose,
+        handleMobileMenuOpen,
+        handleNavigation,
+        getMenuClickHandler,
+        handleProfileMenuOpen,
+        handleSignOut
+    } = useContext(AppContext)
```
Comment (mohamadsoufi) [inline frontend/src/hooks/useAppContext.tsx:31]: this context is doing way tooooo many things, why not have a couple of contexts instead?
examples:
`usePaymentContext`
`useCartContext`

**Review by mohamadsoufi — CHANGES_REQUESTED** [review]

File: frontend/src/contexts/app-context/AppContextTypes.tsx [hunk 2]
```diff
@@ -46,18 +46,18 @@ export type OrderData = {
 
 }
 
-export type userSignin = {
+export type userSignedIn = {
```
Comment (mohamadsoufi) [inline frontend/src/contexts/app-context/AppContextTypes.tsx:49]: ✨

File: frontend/src/hooks/useAppContext.tsx (hunk 1 above)
Comment (Ameer-Alaswad) [reply frontend/src/hooks/useAppContext.tsx:31]: You sure The convention for naming context starts with  -use-  ?

File: frontend/src/components/navbar/UserSettingsMobileMenu.tsx [hunk 3]
```diff
@@ -22,12 +24,14 @@ const UserSettingsMobileMenu: React.FC<UserSettingsMobileMenuProps> = ({ isMobil
     const navigate = useNavigate()
 
     const {
-        handleMobileMenuClose,
+        // handleMobileMenuClose,
```
Comment (mohamadsoufi) [inline frontend/src/components/navbar/UserSettingsMobileMenu.tsx:27]: // 🙈

File: frontend/src/contexts/ContextProviders.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                                <OrdersContextProvider>
+                                    { children }
+                                </OrdersContextProvider>
+                            </UserAuthContextProvider>
+                        </CheckoutContextProvider>
+                    </ShoppingCartContextProvider>
+                </AppContextProvider>
+            </MenuSettingsMobileContextProvider>
+        </MenuSettingsContextProvider>
+    );
```
Comment (mohamadsoufi) [inline frontend/src/contexts/ContextProviders.tsx:28]: ✨

**Review by mohamadsoufi — APPROVED** [review]
AMAZING ✨

**Pull request merged**